### PR TITLE
Changelog update.

### PR DESCRIPTION
**Pull request by cristim**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
+
+I forgot to update this in a while, so this is a quite big changelog entry.
+
+I got the first major code contributions by other developers, so from now on the
+changelog entries will be split by author. It may also have a header like this
+one in which it will contain a short summary or a message from the author.
+
+Special thanks to Hugo Rosnet, who contributed a lot of code that implemented a
+number of major features, helped me with multiple code reviews and kept me
+motivated enough to constantly work on this project.
```
Comment (xlr-8) [inline CHANGELOG.md]: My pleasure :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,50 @@
...
+Github issues.
+
+~Cristian
+
+### Changes by author
+
+#### Cristian Măgherușan-Stanciu
+
+- Big code refactoring to make the code more maintainable and testable.
+- Buildsystem improvements(and regressions, since fixed).
```
Comment (xlr-8) [inline CHANGELOG.md]: space with the first parenthesis

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,50 @@
...
 
+## 29 December 2016, build 158
+
+I forgot to update this in a while, so this is a quite big changelog entry.
+
+I got the first major code contributions by other developers, so from now on the
+changelog entries will be split by author. It may also have a header like this
+one in which it will contain a short summary or a message from the author.
+
+Special thanks to Hugo Rosnet, who contributed a lot of code that implemented a
```
Comment (xlr-8) [inline CHANGELOG.md]: I'd prefer my github nickname rather than explicit first/last name, despite being in my commit messages.

Comment (cristim) [inline CHANGELOG.md]: Done.

**Pull request merged**